### PR TITLE
Add support for minimum compatible version for Extensions and OpenSearch

### LINKED ISSUE #65: [FEATURE] Design/Discuss what goes into extensions.yml
### Is your feature request related to a problem?
Opening this as a follow up of https://github.com/opensearch-project/opensearch-sdk/pull/60#discussion_r930265844

The code reads all of this configuration today. We should get rid of the unnecessary parts of it which were inherited from plugins. 
```
extensions:
  - name: opensearch-sdk
    uniqueId: opensearch-sdk-1
    hostName: 'sdk_host'
    hostAddress: '127.0.0.1'
    port: '4532'
    version: '1.0'
    description: Extension for the Opensearch SDK Repo
    opensearchVersion: '3.0.0'
    javaVersion: '14'
    className: ExtensionsRunner
    customFolderName: opensearch-sdk
    hasNativeController: false	
```

### PR DESCRIPTION
**Pull request by saratvemulapalli**
Signed-off-by: Sarat Vemulapalli <vemulapallisarat@gmail.com>

### Description
Complementing PR to: https://github.com/opensearch-project/OpenSearch/pull/6003
Marking this as draft, while working on #346.

### Issues Resolved
Closes #65 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -102,16 +102,11 @@ A sample `extensions.yml` file is shown below. The `uniqueId` will be used in RE
 extensions:
   - name: hello-world
     uniqueId: opensearch-sdk-java-1
-    hostName: 'sdk_host'
```
Comment (dbwiddis) [inline DEVELOPER_GUIDE.md:105]: Removal here doesn't match DESIGN.md line 50.

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -102,16 +102,11 @@ A sample `extensions.yml` file is shown below. The `uniqueId` will be used in RE
...
     hostAddress: '127.0.0.1'
     port: '4532'
     version: '1.0'
-    description: Extension for the Opensearch SDK Repo
```
Comment (dbwiddis) [inline DEVELOPER_GUIDE.md:109]: Removal here doesn't match DESIGN.md line 53.

**Review by dbwiddis — DISMISSED** [review]

**Review by ryanbogan — APPROVED** [review]

**Comment by ryanbogan** [comment]
Looks like gradle check is failing

**Comment by dbwiddis** [comment]
> Looks like gradle check is failing

Needs companion PR merged first.

**Comment by saratvemulapalli** [comment]
The last snapshot published is on Jan 31[1]. Still waiting for the latest changes in maven. 
https://aws.oss.sonatype.org/content/repositories/snapshots/org/opensearch/opensearch/3.0.0-SNAPSHOT/

**Review by ryanbogan — APPROVED** [review]

**Review by dbwiddis — APPROVED** [review]

**Pull request merged**